src/angular/components/two-factor.component.ts

```diff
@@ -59,8 +63,18 @@ export class TwoFactorComponent implements OnInit, OnDestroy {
             return;
         }
 
+        const queryParamsSub = this.route.queryParams.subscribe((qParams) => {
+            if (qParams.resetMasterPassword != null) {
+                this.resetMasterPassword = qParams.resetMasterPassword;
+            }
+
+            if (queryParamsSub != null) {
+                queryParamsSub.unsubscribe();
+            }
```

Comment:
I am confused as to what this is used for. Why are we controlling this via query string?

Reply:
Takes the `resetMasterPassword` information from the original response in the `sso.component` and passes it along to the `two-factor.component` so the route can be altered on a successful 2fa in the event of 2fa + reset master password flow.

Is there a better way to do this?

Reply:
The 2fa token response should have a property that says you need to resetPassword.

Reply:
Understood - reverted and adjusted conditional.